p2p/discover/udp.go

```diff
@@ -286,6 +342,22 @@ func newUDP(priv *ecdsa.PrivateKey, c conn, natm nat.Interface, nodeDBPath strin
 func (t *udp) close() {
 	close(t.closing)
 	t.conn.Close()
+
+	// close prepared sql statements
```

Comment:
Right. I didn't use defer because the statements are prepared when the UDP server is started (in `newUDP`) and there is a separate logic for closing the server. If I used defer in `newUDP`, wouldn't the statement be closed as soon as the start logic returns? Anyways, I will make this into a generic closing function as suggested, but I will make the change after the merge to avoid further conflicts. I made a TODO list in #15